Update test-plan.md (#3529)
OmniSharp/omnisharp-vscode: test-plan.md

* Completion is available for types that exist in the project (i.e. `Program`)
* Typing `@model DateTime` prompts for completion for the `model` symbol and the `DateTime` symbol.

> #### C# Lightbulbs

Reviewer: ```suggestion
##### C# Lightbulbs
```

> * Typing `@{ var x = new HtmlString("sdf"); }`results in squiggles under `HtmlString`. Selecting `HtmlString` and pressing Ctrl+. results in a change suggestion to fully-qualify that type name as `Microsoft.AspNetCore.Html.HtmlString`. Hitting enter on that suggestion resolves the squiggles and changes the line.

Reviewer 2: You may want to be a little more specific here on which entry to select to future proof ourselves for when we add more light bulbs

Author: I feel like I was super specific, `Microsoft.AspNetCore.Html.HtmlString` is the entire text which is shown by that light bulb. Can you think of a way I could make that wording clearer?

Reviewer 2: ya this is why you don't leave comments later at night. You were totally clear enough lol.

##### C# Diagnostics
* Typing `@ThisDoesNotExist` results in an error being created and squiggled in the .cshtml file. NOTE: This error squiggly will be misaligned due to known issues.

> #### TagHelper Quick Info

Reviewer: ```suggestion
##### TagHelper Quick Info
```

> * Typing `<environment></environment>` then hovering the mouse over the opening tag results in an informative window about that TagHelper.

Reviewer 2: I'd provide an example with a disclaimer that the text may not be exact

Author: ```suggestion
* Typing `<environment></environment>` then hovering the mouse over the opening tag results in a window about that TagHelper that should read something like `EnvironmentTagHelper ITagHelper implementation targeting <environment> elements that conditionally renders content based on the current value of IHostingEnvironment.EnvironmentName...`.
```
